spring-content-commons/src/main/java/org/springframework/content/commons/io/MedializedResource.java

```diff
@@ -0,0 +1,90 @@
+package org.springframework.content.commons.io;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URL;
+
+import org.springframework.core.io.InputStreamResource;
+import org.springframework.core.io.Resource;
+
+public class MedializedResource implements MediaResource {
+
+	private String name;
+	private String mime;
+	private Resource resource;
+
+	public MedializedResource(Resource resource, String mime, String name) {
+		this.resource = resource;
+		this.mime = mime;
+		this.name = name;
+	}
+
+	public void resetResource(Resource r) {
```

Comment:
I dont see this method on the interface, or in fact used anywhere.  Do we really need this?

Reply:
It is used in my Cache Use POC. Standard implementation of spring-cache don't know to handle Resource type, due to necessity to  read its internal stream. But I saw that if I can reset internal resource I can write cache implementation which can handle also Resources. The problem is in cache.put function. The trick is return Resource from cache store  instead original, So I need reset it.

Reply:
My target is to add chache strategy for rendition aspect here in spring-content, as soon as we close this part of code, so we can avoid multiple file transformations.
I found that, the best place is convert method of RenditionProvider. So If you need rendition provider with particular caching needs ( f.e. our AnyToText rendition, there will be many Tesseract text recognition calls). You can annotate that method with @Cachable anotation. Here is example,

@Override
@Cacheable(key ="T(CacheKey).getKey(#fromInputSource, #toMimeType)")
public Resource convert(Resource fromInputSource, String toMimeType) {
...
}

I will make then new PR with interfaces necessary for caching and also with sort of default implementation examples, so we can discuss it.

Why You think we need ResetableResource interface?  We have MediaResource allready. 
Then DefaultMediaResource is Resource in all aspects so any other systems which call spring-content will just handle Resources, so we don't need any sort of config, in my opinion.